build-system/s3-cache-scripts/cache-download.sh

```diff
@@ -17,8 +17,17 @@ function on_exit() {
 # Run on any exit
 trap on_exit EXIT
 
+# Extract endpoint URL if S3_BUILD_CACHE_AWS_PARAMS is set
```

Comment:
not needing aws for this is super useful adding fixes: https://github.com/AztecProtocol/aztec-packages/issues/9503 to the pr description